projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/case-file-view/components/case-file-view-folder/case-file-view-folder.component.ts

```diff
@@ -0,0 +1,244 @@
+  styleUrls: ['./case-file-view-folder.component.scss'],
+  templateUrl: './case-file-view-folder.component.html'
+})
+export class CaseFileViewFolderComponent implements OnInit, OnDestroy {
+
+  private static readonly UNCATEGORISED_DOCUMENTS_TITLE = 'Uncategorised documents';
+
+  @Input() public categoriesAndDocuments: Observable<CategoriesAndDocuments>;
+
+  public nestedTreeControl: NestedTreeControl<DocumentTreeNode>;
+  public nestedDataSource: DocumentTreeNode[];
+  public categories: CaseFileViewCategory[] = [];
+  public categoriesAndDocumentsSubscription: Subscription;
+
+  private getChildren = (node: DocumentTreeNode) => of(node.children);
+  public nestedChildren = (_: number, nodeData: DocumentTreeNode) => nodeData.children;
+
+  constructor() {
+    this.nestedTreeControl = new NestedTreeControl<DocumentTreeNode>(this.getChildren);
+  }
+
+  public ngOnInit(): void {
+    this.categoriesAndDocumentsSubscription = this.categoriesAndDocuments.subscribe(categoriesAndDocuments => {
+      // Using the mock data for now as we have to display the documents as well for demo purpose
+      const categories = this.loadCategories(); // categoriesAndDocuments.categories;
+      // Generate document tree data from categories
+      const treeData = this.generateTreeData(categories);
+      // Append uncategorised documents
+      if (categoriesAndDocuments.uncategorised_documents && categoriesAndDocuments.uncategorised_documents.length > 0) {
+        const uncategorisedDocuments = this.getUncategorisedDocuments(categoriesAndDocuments.uncategorised_documents);
+        treeData.push(uncategorisedDocuments);
+      }
+      // Initialise cdk tree with generated data
+      this.nestedDataSource = treeData;
+    });
+  }
+
+  public generateTreeData(categories: CaseFileViewCategory[]): DocumentTreeNode[] {
+    return categories.reduce((tree, node) => [
+      ...tree,
+      ...[
+        {
+          name: node.category_name,
+          children: [...this.generateTreeData(node.sub_categories), ...this.getDocuments(node.documents)]
+        },
+      ],
+    ], []);
+  }
+
+  public getDocuments(documents: CaseFileViewDocument[]): DocumentTreeNode[] {
+    const documentsToReturn: DocumentTreeNode[] = [];
+    documents.forEach(document => {
+      documentsToReturn.push({ name: document.document_filename });
+    });
+    return documentsToReturn;
+  }
+
+  public getUncategorisedDocuments(uncategorisedDocuments: CaseFileViewDocument[]): DocumentTreeNode {
+    const documents: DocumentTreeNode[] = [];
+    uncategorisedDocuments.forEach(document => {
+      documents.push({ name: document.document_filename });
+    });
+    return { name: CaseFileViewFolderComponent.UNCATEGORISED_DOCUMENTS_TITLE, children: documents };
+  }
+
+  public ngOnDestroy(): void {
+    if (this.categoriesAndDocumentsSubscription) {
+      this.categoriesAndDocumentsSubscription.unsubscribe();
+    }
+  }
+
+  public loadCategories(): CaseFileViewCategory[] {
```

Comment:
Import the test data from the file you've created

Reply:
thank you @MateiHMCTS I will do this as part of the next in progress ticket https://tools.hmcts.net/jira/browse/EUI-5377